### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by isomethane**

### CODE REVIEW

File: .github/workflows/qodana.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Qodana - Code Inspection
```
Comment (nbirillo) [inline .github/workflows/qodana.yml]: Do we really need Qodana action? For the Reflekt project, a special build on team city was set up to visualize the report. In this case, this is just an extra check.

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    repositories {
+        jcenter()
+        google()
+    }
+
+    // We should publish the project in the local maven repository before the tests running
+    tasks.withType<Test> {
+        dependsOn(tasks.withType<PublishToMavenLocal>{})
+    }
```
Comment (nbirillo) [inline build.gradle.kts]: Why? In Reflekt project we used Reflekt DSL in tests. Probably we can add it later if it is necessary in the future.

File: examples/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+tasks.withType<KotlinJvmCompile> {
+    kotlinOptions {
+        jvmTarget = "11"
+        languageVersion = "1.5"
+        apiVersion = "1.5"
+    }
+}
+
```
Comment (nbirillo) [inline examples/build.gradle.kts:19]: Should we add a section with the kinference-compiler plugin (with enabled option)?

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great, thank you!

File: build.gradle.kts (hunk 2 above)
Comment (isomethane) [reply build.gradle.kts]: Well, I'm sure we'll need it in the nearest future, but I agree that adding dependencies before we actually need them is a bad idea, thank you.

**Review by nbirillo — APPROVED** [review]
Great, thank you!

**Pull request merged**